src/services/utility.service.ts

```diff
@@ -1,10 +1,35 @@
-import { Range, ZoomLevel, SearchResult } from "../types/main";
+import {
+  Range,
+  ZoomLevel,
+  SearchResult,
+  Location,
+  MarkerLocation,
+} from "../types/main";
 import {
   Buildings,
   POIToSearchResult,
   buildingToSearchResult,
   POIInfo,
 } from "../constants";
+import { CURRENT_LOCATION_DISPLAY_TEXT } from "../styles";
+
+/**
+ * Converts a given MarkerLocation into a search result.
+ *
+ * A search result is a representation of a MarkerLocation used for the
+ * autocomplete component, and can be a building, POI or campus.
+ *
+ * @param markerLocation MarkLocation to be converted
+ */
+const MarkerToSearchResult = (markerLocation: MarkerLocation): SearchResult => {
+  return {
+    id: markerLocation.id,
+    displayName: markerLocation.displayName,
+    location: markerLocation.location,
+    searchName: markerLocation.displayName,
+    extraInformation: "",
+  };
+};
```

Comment:
Please move this to be inside the class, as a private helper function

Reply:
Made it into a public helper function and it's used in 2 files.